src/gt4py/next/otf/compilation/compiler.py

```diff
@@ -67,17 +68,18 @@ def __call__(
     ) -> stages.ExtendedCompiledProgram:
         src_dir = cache.get_cache_folder(inp, self.cache_lifetime)
 
-        data = build_data.read_data(src_dir)
+        with lock.Lock(str(src_dir / "file.lock"), lifetime=600):  # type: ignore[attr-defined] # mypy not smart enough to understand custom export logic
```

Comment:
```suggestion
        with lock.Lock(str(src_dir / "compilation.lock"), lifetime=600):  # type: ignore[attr-defined] # mypy not smart enough to understand custom export logic
```
Maybe something less generic then file?